qiskit_experiments/base/__init__.py

```diff
@@ -0,0 +1,88 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+========================================================
+Experiment Base Classes (:mod:`qiskit_experiments.base`)
+========================================================
+
+.. currentmodule:: qiskit_experiments.base
+
+Classes
+=======
+
+.. autosummary::
+    :toctree: ../stubs/
+
+    BaseExperiment
+    BaseAnalysis
+    Options
+
+Creating Experiments
+====================
+
+Experiments and analysis of experiment data is done by subclassing the
+:class:`BaseExperiment` and :class:`BaseAnalysis` classes.
+
+Experiment Subclasses
+*********************
+
+To create an experiment subclass
+
+- Implement the abstract :meth:`BaseExperiment.circuits` method.
+  This should return a list of ``QuantumCircuit`` objects defining
+  the experiment payload.
+- Call the :meth:`BaseExperiment.__init__` method during the subclass
+  constructor with a list of physical qubits. The length of this list must
+  be equal to the number of qubits in each circuit and is used to map these
+  circuits to this layout during execution.
+- Set :attr:`BaseExperiment.__analysis_class__` class attribute to
+  specify the :class:`BaseAnalysis` subclass for analyzing result data.
+
+Optionally the following methods can also be overridden in the subclass to
+allow configuring various experiment and execution options
+
+- :meth:`BaseExperiment._default_experiment_options`
+  to set default values for configurable option parameters for the experiment.
+- :meth:`BaseExperiment._default_analysis_options`
```

Comment:
I don't think `_default_analysis_options` can be overridden. If we override this method, we could accidentally induce a mismatch between kwargs that are expected by the analysis class and actually the experiment class provides. This also causes a trouble in auto documentation #116

Reply:
You can do it, but yes it could lead to bugs. The main reason would be if you had 1 analysis class used by multiple experiments and needed to define different defaults for each experiment.

Reply:
Can you describe this here? This seems like recommending user to override analysis default of experiment. This should be limited to the situation you mentioned, and user should care the mismatch of run analysis API (= we should update defaults of analysis class unless one have special reason to do that)